tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -129,7 +131,9 @@ private Security(SecuritySettings settings = null, InstrumentationGateway instru
 
                     _instrumentationGateway.EndRequest += ReportWafInitInfoOnce;
                     LifetimeManager.Instance.AddShutdownTask(RunShutdown);
-                    _rateLimiter = new RateLimiterTimer(_settings.TraceRateLimit);
+                    _rateLimiter = new SpanCounterKeeper(_settings.TraceRateLimit);
+                    _cancellationTokenSource = new CancellationTokenSource();
+                    RunRateLimitLoop(TimeSpan.FromSeconds(1), _cancellationTokenSource.Token);
```

Comment:
Shouldn't you have a `Task.Run()` here?

Reply:
No, because it's async void, there's no task to run.